### PR TITLE
Improve project

### PR DESCRIPTION
**Pull request by OsPit01**

### CODE REVIEW

File: src/Main.java [hunk 1]
```diff
@@ -19,16 +19,22 @@ public static void main(String[] args) {
                     String inputPassword = scanner.nextLine();
                     LoginCommand loginCommand = new LoginCommand();
                     loginCommand.execute(inputName, inputPassword);
+
```
Comment (VladOsipov) [inline src/Main.java]: the same style everywhere ( other cases without empty line)

File: src/UserException.java [hunk 2]
```diff
@@ -0,0 +1,21 @@
+public class UserException extends Exception {
```
Comment (VladOsipov) [inline src/UserException.java]: Review existing exception and their names.
We can ussually understand why this exception was thrown based on its name

Your exception don't give any info about an exception

File: src/RegisterCommand.java [hunk 3]
```diff
@@ -4,5 +4,7 @@ public void register(String userName, String userPassword) {
 
         UserContainer.getUsers().put(userName, userPassword);
         System.out.println("Registration was successful");
+        System.out.println("input your name");
```
Comment (VladOsipov) [inline src/RegisterCommand.java]: looks redundant

File: src/User.java [hunk 4]
```diff
@@ -0,0 +1,11 @@
+import java.util.ArrayList;
+
+public class User {
+    ArrayList<String> name = new ArrayList<>();
```
Comment (VladOsipov) [inline src/User.java]: Change type to List.
We should use interfaces as much as we can ( instead of implementations)

Comment (VladOsipov) [inline src/User.java]: String name;
List<String> names;

plural form for collections and maps

**Review by VladOsipov — CHANGES_REQUESTED** [review]

File: src/Main.java (hunk 1 above)
Comment (OsPit01) [reply src/Main.java]: where ?

Comment (VladOsipov) [reply src/Main.java:22]: Each line before `break` is not empty:
line 37
line 42
line 47

**Review by VladOsipov — CHANGES_REQUESTED** [review]

File: src/Main.java (hunk 1 above)
Comment (VladOsipov) [reply src/Main.java]: Each line before `break` is not empty:
line 37
line 42
line 47

File: src/User.java [hunk 5]
```diff
@@ -2,14 +2,13 @@
 import java.util.List;
 
 public class User {
-
- static List<String> infoUser = new ArrayList<>();
     String name;
```
Comment (VladOsipov) [inline src/User.java]: private/public?

File: src/User.java [hunk 6]
```diff
@@ -2,14 +2,13 @@
...
     String surname;
+    int id;
     String login;
     String password;
 
-    public void addInfo(String name, String surname, String login, String password) {
+    public void addUser(String name, String surname, String login, String password) {
```
Comment (VladOsipov) [inline src/User.java]: user.addUser() ? what does it mean?
if you want to add data to user you have 2 options:
- Add this data via constructor
- Use setters

File: src/User.java [hunk 7]
```diff
@@ -32,13 +31,11 @@ public String getPassword() {
...
-    public void addInArray() {
-        String result = "username:" + " " + getName() + " " + "surname:" + " " + getSurname() + " " + "login:" + getLogin() + " " + "password:" + getPassword();
-         infoUser.add(result);
-        System.out.println(infoUser);
-
-       }
+    public int getId() {
+        return id;
     }
+   // public void call() {
```
Comment (VladOsipov) [inline src/User.java]: ???

File: src/UserContainer.java [hunk 8]
```diff
@@ -17,4 +28,6 @@ public static boolean isUserNameAndPasswordExists(String userName, String passwo
         return userExists;
     }
 
+
+
```
Comment (VladOsipov) [inline src/UserContainer.java:30]: 2 empty lines

File: src/UserContainer.java [hunk 9]
```diff
@@ -1,9 +1,20 @@
...
+    static ArrayList<User> infoUser = new ArrayList();
    private final static Map<String, String> users = new HashMap<String, String>();
 
+public void containerSave() {
+    User user = new User();
+    infoUser.add(user);
+
+}
+
+
```
Comment (VladOsipov) [inline src/UserContainer.java:14]: 3 empty lines

File: src/UserContainer.java [hunk 10]
```diff
@@ -1,9 +1,20 @@
+import java.util.ArrayList;
 import java.util.HashMap;
 import java.util.Map;
 
 public class UserContainer {
+
+    static ArrayList<User> infoUser = new ArrayList();
    private final static Map<String, String> users = new HashMap<String, String>();
 
+public void containerSave() {
```
Comment (VladOsipov) [inline src/UserContainer.java]: this method creates new (empty) user and saves it to list?
Why do you need this

File: src/UserContainer.java [hunk 11]
```diff
@@ -1,9 +1,20 @@
+import java.util.ArrayList;
 import java.util.HashMap;
 import java.util.Map;
 
 public class UserContainer {
+
+    static ArrayList<User> infoUser = new ArrayList();
```
Comment (VladOsipov) [inline src/UserContainer.java]: pruvate/public?

**Review by VladOsipov — CHANGES_REQUESTED** [review]

File: src/User.java (hunk 7 above)
Comment (OsPit01) [reply src/User.java]: here was be sout

**Pull request closed without merging**